pages/heft_tasks/typescript.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: page
+title: '"typescript" task'
+navigation_source: docs_nav
+---
+
+This task invokes the compiler for the [TypeScript]https://www.typescriptlang.org/) programming language.
+
+
+## When to use it
+
+TypeScript is the standard programming language for Rush Stack.  There are many benefits to using one "lingua franca" across all coding investments, rather than having to maintain different expertise and libraries for different languages.
+
+In particular, we recommend TypeScript for:
+
+- **Application development:** It's even a good choice for prototypes and small experiments.
+- **Tooling projects:** A great developer experience multiplies everyone's productivity, so build tools should be treated as first-class software projects with their own designs, documentation, and tests.
+- **Native apps:** Where possible, TypeScript can also be used for native development via runtime layers such as [Electron](https://www.electronjs.org/) and [React Native](https://reactnative.dev/).
+
+
+## Config files
+
+The primary configuration comes from TypeScript's [tsconfig.json](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html) file.
+
+The current implementation relies on the `@microsoft/rush-stack-compiler-*` packages, which bundle together the TypeScript compiler package, common associated dependencies and tools, and shared config files. The **rush-stack-compiler** system enables monorepo projects to easily switch between different TypeScript compiler releases, while ensuring that those pieces will be a compatible combination that has been tested.
+
+> The `@microsoft/rush-stack-compiler-*` system is inherited from the earlier [gulp-core-build](https://github.com/microsoft/rushstack/tree/master/core-build/gulp-core-build) toolchain which preceded Heft.  Eventually we plan to replace it with a more lightweight Heft-specific equivalent.
+
+To use this system, your **tsconfig.json** file should extend from one of the `includes/tsconfig-*.json` templates.  For example, to target TypeScript 3.9 you would add `@microsoft/rush-stack-compiler-3.9` to `devDependencies` in your **package.json** file, and then reference it like this:
```

Comment:
It's also useful to mention that Heft will pick up and use any file called `/tsconfig-.+\.json/` in the project root. So you can have a `tsconfig.json` and a `tsconfig-web.json` for a Node project that needs to emit both ES6 for Node and ES5 for browser.

Reply:
🤔 @iclanton Let's chat about this separately. Detecting `tsconfig` patterns seems potentially confusing. Since it's a somewhat advanced scenario, it might be better to explicitly register them in Heft's **typescript.json** file.